cfme/tests/infrastructure/test_provisioning_rest.py

```diff
@@ -73,6 +74,20 @@ def provision_data(appliance, provider, small_template_modscope):
     return get_provision_data(appliance.rest_api, provider, small_template_modscope.name)
 
 
+def clean_vm(provider, vm_name):
+    if not provider.mgmt.is_vm_stopped(vm_name):
+        provider.mgmt.stop_vm(vm_name)
+        provider.mgmt.wait_vm_stopped(vm_name)
+    provider.mgmt.delete_vm(vm_name)
+
+
+def request_finished(provision_request):
```

Comment:
wait_for + https://github.com/lkhomenk/integration_tests/blob/ffac0136cf5bc87b457a334094dbd29d838ee800/cfme/services/requests.py#L186 also might be useful. it also uses REST by default

Reply:
This is also checking the task state, but that is missing from this test anyway from the looks of it.  @jan-zmeskal please investigate and consider this change to using Requests methods.